core_lib/src/graphics/bitmap/bitmapimage.cpp

```diff
@@ -650,6 +650,44 @@ void BitmapImage::drawPath(QPainterPath path, QPen pen, QBrush brush,
     modification();
 }
 
+int BitmapImage::findLeft(QRectF rect, int grayValue)
+{
+    int left = static_cast<int>(rect.left());
+    int right = static_cast<int>(rect.right());
+    int top = static_cast<int>(rect.top());
+    int bottom = static_cast<int>(rect.bottom());
+    for (int x = left; x <= right; x++)
+    {
+        for (int y = top; y <= bottom; y++)
+        {
+            if (qGray(constScanLine(x,y)) < grayValue)
+            {
+                return x;
+            }
+        }
+    }
+    return 10000;
```

Comment:
Should return -1 here instead of 10000.

Reply:
The function 'findLeft' searches for, and returns, the left-most x-value of the peg bar you have selected. This value could in some cases be -1, which is pretty close to the center of the canvas. That's why I chose 10000 as a return value.

Reply:
Hmm... in that case I'd prefer if we could maybe settle on another return type. Given that the coordinate can be anything, even 10000, returning a custom value that can hold both the intended value but also a potential Errorcode would be better and easier to understand for people trying to read the code too.

So it could be something like... 
```
... will be defined in pencilerror.h
struct StatusInt {
        int value = 0;
        ErrorCode errorcode;
    };

and then return that value in your findLeft and findTop top function.

Now you can simply return a StatusInt and catch the appropriate errorcode instead.
This could come in handy in many other places, so I think that would be a nice addition to the project.

Reply:
I have never made a struct before, but I can see that it often is defined as a class.
However, I can't get it to work unless I put it inside the Status class, after the ErrorCode Enum, like this:
```
        ERROR_NEED_AT_LEAST_ONE_CAMERA_LAYER
    };

    struct StatusInt {
        int value = 0;
        ErrorCode errorcode;
    };

```
Is that the way to do it? It must somehow know what ErrorCode is?

Reply:
Yeah that's correct, the struct needs to the origin of the values inside it. Putting the struct above Errorcode isn't allowed because ErrorCode doesn't exist yet.